CMakeLists.txt

```
@@ -1,26 +1,73 @@
 cmake_minimum_required(VERSION 3.25)
 
 # Set project name
-project(tracker VERSION 1.0 LANGUAGES CXX)
+project(tracker VERSION 1.0 LANGUAGES Swift OBJC CXX)
 
-# Set Binary Dir in project
-set(CMAKE_BINARY_DIR build/bin)
+if(IOS)
+  set(CMAKE_OSX_DEPLOYMENT_TARGET "15.0" CACHE INTERNAL "")
+else()
+  set(CMAKE_OSX_DEPLOYMENT_TARGET "11.0" CACHE INTERNAL "")
+endif()
 
-add_executable(tracker src/main.cpp)
+# If we are compiling for macOS we want to target OS versions down to 11.0
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Werror")
 
-add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/external/tracktion_engine/modules/juce ${CMAKE_BINARY_DIR}/tracktion/juce EXCLUDE_FROM_ALL)
-add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/external/tracktion_engine/modules ${CMAKE_BINARY_DIR}/tracktion EXCLUDE_FROM_ALL)
+# Swift related settings
+set(CMAKE_Swift_LANGUAGE_VERSION 5)
+set(SWIFT_OPTIMIZATION_LEVEL "-Onone") # Required for preview feature
+                                       # Compile without any optimization
 
-add_library(tracktion STATIC)
+# Adds all the module sources so they appear correctly in the IDE
+set_property(GLOBAL PROPERTY USE_FOLDERS YES)
+option(JUCE_ENABLE_MODULE_SOURCE_GROUPS "Enable Module Source Groups" ON)
+option(JUCE_BUILD_EXTRAS "Build JUCE Examples" OFF)
+option(JUCE_BUILD_EXAMPLES "Build JUCE Extras" OFF)
+add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/external/tracktion_engine/modules/juce)
+add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/external/tracktion_engine/modules ./cmake_build_tracktion)
 
-target_link_libraries(tracktion 
+###############################
+# Create Tracktion Static Lib #
+###############################
+add_library(Tracktion STATIC)
+target_include_directories(Tracktion INTERFACE ${CMAKE_CURRENT_LIST_DIR}/external/tracktion_engine/modules
```

```suggestion
target_include_directories(Tracktion PUBLIC ${CMAKE_CURRENT_LIST_DIR}/external/tracktion_engine/modules
```
Let's try PUBLIC here